packages/shared-ux/file/file_picker/impl/src/file_picker.tsx

```diff
@@ -31,14 +31,18 @@ import { EmptyPrompt } from './components/empty_prompt';
 import { FileGrid } from './components/file_grid';
 import { SearchField } from './components/search_field';
 import { ModalFooter } from './components/modal_footer';
-
 import { ClearFilterButton } from './components/clear_filter_button';
+import { DeletePrompt } from './components/delete_prompt/delete_prompt';
 
 export interface Props<Kind extends string = string> {
   /**
    * The file kind that was passed to the registry.
    */
   kind: Kind;
+  /**
+   * A function which determines whether to show a delete button for a file.
+   */
+  deleteButtonPredicate?: (file: FileJSON) => boolean;
```

Comment:
`deleteButtonPredicate` -> `allowDelete` or `shouldAllowDelete` ? 

as `predicate` doesn't explain what it is the condition for. 

also, I won't specify that there is a `button`. the UI implementation for deleting an image could be anything and the prop could still be valid for it